test/client/http_service.test.js

```diff
@@ -0,0 +1,265 @@
+const http = require('http');
+const StitchMongoFixture = require('../fixtures/stitch_mongo_fixture');
+
+import { buildClientTestHarness, extractTestFixtureDataPoints } from '../testutil';
+
+describe('With a mock server', () => {
+  const TEST_SERVER_URL = 'http://localhost:7000';
+
+  let handleRequest;
+  const server = http.createServer((req, res) => handleRequest(req, res));
+
+  beforeAll(() => server.listen(7000));
+  afterAll(() => server.close());
+
+  describe('And http service functions', () => {
```

Comment:
let's test a few more failure cases, also.
- no matching rules
- bad args (invalid URL for instance, or missing a required option, etc.)

these should trigger `.catch()` on the SDK request.